work with array

Anna190997 opened the pull request.

Anna190997 committed:
work with array

PolinaShalak commented on tests/palindrome.test.js:
```diff
@@ -0,0 +1,10 @@
+  expect(palindrome('level')).toBe(true);
+});
+
+import { search } from '../src/main.js';
```
Все импорты нужно выносить в самых верх файлика

PolinaShalak commented on src/main.js:
```diff
@@ -1,9 +1,185 @@
+  delete hotel.country;
+  arr[key] = hotel;
+  return arr;
+}, {});
```
В дз нужно возвращать массив с городами, а получается объект с ключом city в котором расположен массив
![image](https://user-images.githubusercontent.com/71837845/169091422-eb37b09d-a6f2-40d8-8706-669a40ed58b3.png)

PolinaShalak replied:
И должна быть функция, которая принимает в себя массив и возвращает результат: 
const getContries = (array) => {
  // тело функции
}

PolinaShalak commented on src/main.js:
```diff
@@ -1,9 +1,185 @@
+hotels.forEach(function (a) {
+  if (!this[a.country]) {
+    this[a.country] = { city: [], country: a.country };
+    grouped.push(this[a.country]);
```
Можно попробовать написать намного проще, сделать 1 reduce, результат которого будет являться объектом, при каждой итерации добавлять этому объекту ключ - страну, и значение объекта - город
 return ({[country]: [...object?.[country]?.city, city]})
 То естьвозвращаем города, которые уже были добавлены по этому ключу, и записываем новый, который получили в результате итерации

Anna190997 committed:
fixed uniqueCountries
Сделать получилось, но все равно длинным путем, так как ты писала что-то не выходит

PolinaShalak committed:
Merge branch 'main' into lesson-6

# Conflicts:
#	src/main.js
